.kiro/specs/ai-tutor-rag/design.md

```
@@ -0,0 +1,882 @@
+        EF_Chat->>EF_Chat: Assemble system prompt (persona + CLO context + chunks)
+        EF_Chat->>OpenRouter: Stream completion request
+        OpenRouter-->>EF_Chat: Token stream (SSE)
+        EF_Chat-->>ChatUI: Forward SSE token stream
+        EF_Chat->>DB: Persist user + assistant messages with citations
+        EF_Chat->>DB: Update usage counters (messages, tokens)
+    end
+```
+
+### Component Architecture
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)